src/server.c

```diff
@@ -3461,19 +3461,17 @@ int processCommand(client *c) {
         return C_ERR;
     }
 
+    /* Set server.script_oom to 0, if we will find out that we are
+     * over the memory limit, it will be set to 1. */
+    server.script_oom = 0;
```

Comment:
that will break if the script gets BUSY and starts processing other commands (it could override this variable).

i think it must have a `!scriptIsTimedout()` condition